pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,183 @@
+package nodelabeller
+
+import (
+	goerror "errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"
+)
+
+const (
+	labelNamespace    = "feature.node.kubernetes.io"
+	labellerNamespace = "node-labeller"
+)
+
+var nodeLabellerVolumePath = "/var/lib/kubevirt-node-labeller"
+
+//NodeLabeller struct holds informations needed to run node-labeller
+type NodeLabeller struct {
+	kvmController     *device_manager.DeviceController
+	configMapInformer cache.SharedIndexInformer
+	clientset         kubecli.KubevirtClient
+	host              string
+	namespace         string
+}
+
+func NewNodeLabeller(kvmController *device_manager.DeviceController, configMapInformer cache.SharedIndexInformer, clientset kubecli.KubevirtClient, host, namespace string) *NodeLabeller {
+	return &NodeLabeller{
+		kvmController:     kvmController,
+		configMapInformer: configMapInformer,
+		clientset:         clientset,
+
+		host:      host,
+		namespace: namespace,
+	}
+}
+
+//Run runs node-labeller
+func (n *NodeLabeller) Run(stop chan struct{}) {
+	logger := log.DefaultLogger()
+	logger.Infof("node-labeller is running")
+
+	n.configMapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			err := n.run()
```

Comment:
we shouldn't be executing logic like this in the informer callbacks. Instead we should create a workqueue and pop a key on to the queue to handle performing the logic.